src/App.jsx

```diff
@@ -1,56 +1,128 @@
-import { initialColors } from "./lib/colors";
+import { initialThemes, initialColors } from "./lib/colors";
 import { uid } from "uid";
 import useLocalStorageState from "use-local-storage-state";
-import Color from "./Components/Color/Color";
-import ColorForm from "./Components/ColorForm/ColorForm";
+import Themes from "./Components/Themes/Themes";
 import "./App.css";
+import { ThemesMenu } from "./Components/ThemesMenu/ThemesMenu";
+import { useState } from "react";
 
 function App() {
+  const [themes, setThemes] = useLocalStorageState("themes", {
+    defaultValue: initialThemes,
+  });
+
+  const [currentTheme, setCurrentTheme] = useState(themes[0]);
+
   const [colors, setColors] = useLocalStorageState("colors", {
     defaultValue: initialColors,
   });
 
+  function updateCurrentTheme(updatedTheme) {
+    const updatedThemes = themes.map((theme) =>
+      theme.id === currentTheme.id ? updatedTheme : theme
+    );
+
+    setThemes(updatedThemes);
+    setCurrentTheme(updatedTheme);
+  }
+
   function handleAddColor(newColor) {
-    setColors([{ id: uid(), ...newColor }, ...colors]);
+    const updatedColor = { id: uid(), ...newColor };
+    setColors([updatedColor, ...colors]);
+    if (currentTheme.id === "t1") {
+      return;
+    } else {
+      const updatedTheme = {
+        ...currentTheme,
+        colors: [updatedColor, ...currentTheme.colors],
+      };
+      updateCurrentTheme(updatedTheme);
+    }
   }
 
   function handleDeleteColor(id) {
-    const colorsToKeep = colors.filter((color) => {
-      return color.id !== id;
-    });
-    setColors(colorsToKeep);
+    if (currentTheme.id === "t1") {
+      return;
+    } else {
+      const updatedColors = currentTheme.colors.filter(
+        (color) => color.id !== id
+      );
+      const updatedTheme = { ...currentTheme, colors: updatedColors };
+      updateCurrentTheme(updatedTheme);
+    }
   }
 
   function handleEditColor(id, newColor) {
-    const newEditColor = colors.map((color) => {
-      return color.id === id ? { ...color, ...newColor } : color;
-    });
-    setColors(newEditColor);
+    if (currentTheme.id === "t1") {
+      return;
+    } else {
+      const updatedColors = currentTheme.colors.map((color) =>
+        color.id === id ? { ...color, ...newColor } : color
+      );
+
+      const updatedTheme = { ...currentTheme, colors: updatedColors };
+      updateCurrentTheme(updatedTheme);
+    }
+  }
+
+  function handleAddTheme(newThemeName) {
+    const newTheme = { id: uid(), name: newThemeName, colors: [] };
+    setThemes([...themes, newTheme]);
+    setCurrentTheme(newTheme);
```

Comment:
you can now here add :  ` setIsAddingTheme(false);` ......so if the user added a theme ,the interface will turn to the dropdown menu and no longer showing the input field